sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1434,7 +1434,9 @@ class DataFrame private[sql](
     }
   }
 
-  private[sql] def internalRowRdd = queryExecution.executedPlan.execute()
+  // Note that the RDD is memoized. Once called, it won't change even if you change any
+  // query planning related Spark SQL configurations.
+  private[sql] def internalRowRdd = queryExecution.toRdd
```

Comment:
This patch change the behavior of `internalRowRdd `(as the comment says), is this what we want?

If it's, I think making it `lazy val` will be more clear than the comment says.